core-library/src/main/java/org/silverpeas/core/admin/space/SpaceInst.java

```diff
@@ -319,12 +313,24 @@ public String getFirstPageExtraParam() {
   }
 
   public List<SpaceInst> getSubSpaces() {
-    return List.copyOf(subSpaces);
+    return data.safeRead((p, s, c) -> List.copyOf(s));
+  }
+
+  public void setData(final List<SpaceProfileInst> profiles, final List<SpaceInst> subSpaces,
+      final List<ComponentInst> components) {
+    data.reset();
+    data.manualWrite((p, s, c) -> {
+      p.addAll(profiles);
+      s.addAll(subSpaces);
+      c.addAll(components);
+    });
```

Comment:
Here, accordingly to the implementation, the data aren't set but more added to the existing ones. So, Shouldn't the setData method be renamed to addData so to avoid misunderstanding?

Reply:
The aim of setData method is effectively to perform a set and not an add.
Indeed, data.reset() is called just before the manualWrite method.

Reply:
Ha yes